autotx/utils/ethereum/lifi/swap.py

```diff
@@ -168,23 +168,85 @@ def build_swap_transaction(
                 }
             )
             transactions.append(
-                models.ApproveTransaction.create(
-                    token_symbol=token_in_symbol,
-                    token_address=str(token_in_address),
+                ApproveTransaction.create(
+                    token=Token(symbol=token_in_symbol, address=str(token_in_address)),
                     amount=float(Decimal(str(quote.amount_in)) / 10 ** token_in_decimals),
                     spender=approval_address,
                     params=cast(dict[str, Any], tx),
                 )
             )
     transactions.append(
-        models.SwapTransaction.create(
-            from_token_symbol=token_in_symbol,
-            to_token_symbol=token_out_symbol,
-            from_token_address=str(token_in_address),
-            to_token_address=str(token_out_address),
+        SwapTransaction.create(
+            from_token=Token(symbol=token_in_symbol, address=str(token_in_address)),
+            to_token=Token(symbol=token_out_symbol, address=str(token_out_address)),
             from_amount=float(Decimal(str(quote.amount_in)) / 10 ** token_in_decimals),
             to_amount=float(Decimal(str(int(quote.to_amount_min))) / 10 ** token_out_decimals),
             params=cast(dict[str, Any], quote.transaction),
         )
     )
     return transactions
+
+def can_build_swap_transaction(
```

Comment:
what's the purpose of this function? doesn't it do the same as `build_swap_transaction` with the difference that it doesn't return an array of transactions?

Reply:
Exactly, the idea is to check if we can build the swap transactions. If we can we can proceed.
We don't want to actually use the built transactions since we only care about intents at this stage